### ISSUE TITLE
fail the build if a retracted version is present in go.mod

### ISSUE THREAD

**Issue by kevinburkesegment** [golangci/golangci-lint#5855]
### Welcome

- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

go install

### Your feature request related to a problem? Please describe

It's bad if people import versions of software that the software authors don't think they should import, and are concerned enough about that they've published retractions. 

The UX to do this directly with the Go tool is clunky: https://github.com/golang/go/issues/73952#issuecomment-2932519089

### Describe the solution you'd like

Specifically, golangci-lint should offer a checker that searches for retracted versions of dependencies imported by the current project, and fails the build if it finds any. 

### Describe alternatives you've considered

One was suggested in the thread, but involves four lines of Bash programming

```bash
retracted="$(go list -m -retracted -f '{{ if .Retracted }}{{ .Path }}: {{ .Retracted }}{{ end }}' all)"
if [[ -n "${retracted}" ]]; then
  # TODO: format a nice message and send ${retracted} to build summary
  exit 1
fi
```

### Supporter

- [ ] I am a sponsor through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
This is outside the scope of golangci-lint: this is not a static analysis linter.

I recommend creating a plugin: https://golangci-lint.run/plugins/module-plugins/
